cmd/dvotecli/commands/file.go

```diff
@@ -0,0 +1,145 @@
+package commands
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"go.vocdoni.io/dvote/client"
+	"go.vocdoni.io/dvote/types"
+	"nhooyr.io/websocket"
+)
+
+var fileCmd = &cobra.Command{
+	Use:   "file",
+	Short: "file subcommands",
+}
+
+var fileAddCmd = &cobra.Command{
+	Use:   "add [name] [base64 payload]",
+	Short: "add file from a Base64 payload",
+	RunE:  fileAdd,
+}
+
+var fileFetchCmd = &cobra.Command{
+	Use:   "fetch [ipfs://<hash>]",
+	Short: "fetch file from URI, Base64 encoded",
+	RunE:  fileFetch,
+}
+
+var pinListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "list pinned files",
+	RunE:  pinList,
+}
+
+func init() {
+	rootCmd.AddCommand(fileCmd)
+	fileCmd.AddCommand(fileAddCmd)
+	fileCmd.AddCommand(fileFetchCmd)
+	fileCmd.AddCommand(pinListCmd)
+}
+
+func fileAdd(cmd *cobra.Command, args []string) error {
+
+	if err := checkSignKey(); err != nil {
+		return err
+	}
+
+	if len(args) < 2 {
+		return fmt.Errorf("you must provide a file name and payload")
+	}
+
+	cl, err := client.New(host)
+	if err != nil {
+		return err
+	}
+	defer cl.Conn.Close(websocket.StatusNormalClosure, "")
+
+	var req types.MetaRequest
+	req.Method = "addFile"
+	req.Type = "ipfs"
+	req.Content, err = base64.StdEncoding.DecodeString(args[1])
+	if err != nil {
+		return err
+	}
+	req.Name = args[0]
+	resp, err := cl.Request(req, signKey)
+	if err != nil {
+		return err
+	}
+	if !resp.Ok {
+		return fmt.Errorf(resp.Message)
+	}
+	fmt.Printf("URI: %v", resp.URI)
+
+	return nil
+}
+
+func fileFetch(cmd *cobra.Command, args []string) error {
+
+	if err := checkSignKey(); err != nil {
+		return err
+	}
+
+	if len(args) < 1 {
+		return fmt.Errorf("you must provide a file URI")
+	}
+
+	cl, err := client.New(host)
+	if err != nil {
+		return err
+	}
+	defer cl.Conn.Close(websocket.StatusNormalClosure, "")
+
+	var req types.MetaRequest
+	req.Method = "fetchFile"
+	req.URI = args[0]
+	resp, err := cl.Request(req, signKey)
+	if err != nil {
+		return err
+	}
+	if !resp.Ok {
+		return fmt.Errorf(resp.Message)
+	}
+	b64content := base64.StdEncoding.EncodeToString(resp.Content)
+	fmt.Printf("%v", b64content)
+
+	return nil
+}
+
+func pinList(cmd *cobra.Command, args []string) error {
+
+	if err := checkSignKey(); err != nil {
+		return err
+	}
+
+	cl, err := client.New(host)
+	if err != nil {
+		return err
+	}
+	defer cl.Conn.Close(websocket.StatusNormalClosure, "")
+
+	// Increase the default response size for large lists
+	cl.Conn.SetReadLimit(32768 * 16)
+
+	var req types.MetaRequest
+	req.Method = "pinList"
+	resp, err := cl.Request(req, signKey)
+	if err != nil {
+		return err
+	}
+	if !resp.Ok {
+		return fmt.Errorf(resp.Message)
+	}
+	var pins *map[string]string
```

Comment:
Do this really works? Usually a map is always declared using `make(map[string]string)`, else there will be a null pointer reference when trying to access it. Maybe `json.Unmarshal()` does the make but we need to be sure.